src/hera/workflows/_context.py

```diff
@@ -23,15 +23,27 @@ def __init__(self) -> None:
 
     def enter(self, p: Subbable) -> None:
         if not isinstance(p, Subbable):
-            raise InvalidType()
+            raise InvalidType(type(p))
         self._pieces.append(p)
 
     def exit(self) -> None:
         self._pieces.pop()
 
     def add_sub_node(self, node: Union[SubNodeMixin, TTemplate]) -> None:
         if self._pieces:
-            self._pieces[-1]._add_sub(node)
+            try:
```

Comment:
This is worthy of review

Reply:
@flaviuvadan it would be great to have some comments explaining these steps for future contributions.